frontend/src/components/modules/Tournaments/TournamentListing/TournamentsList.tsx

```diff
@@ -34,7 +35,18 @@ const TournamentList: React.FC = () => {
     }
   }, [currentTab]);
 
-  const tournamentsToRender = currentTab === defaultTab ? ongoing : upcoming;
+  const tournamentsToRender = (): Tournament[] => {
+    switch (currentTab) {
+      case "past":
+        return past;
+      case "ongoing":
+        return ongoing;
+      case "upcoming":
+        return upcoming;
+      default:
+        return ongoing;
+    }
+  };
```

Comment:
I might prefer having an object that maps tabTypes to the corresponding tabs but ok.
    return object[currentTab] || ongoing;